### ISSUE TITLE
Error with python train_RBM.py -d 784,500

### ISSUE THREAD

**Issue by cossio** [atreyasha/deep-generative-models#2]
Trying to execute

```
python train_RBM.py -d 784,500
```

results in the following error:

```
Epoch: 1
Batch number: 0/12000
Traceback (most recent call last):
  File "/home/cossio/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/gen_math_ops.py", line 3606, in equal
    "incompatible_shape_error", incompatible_shape_error)
tensorflow.python.eager.core._FallbackException: This function does not handle the case of the path where all inputs are not already EagerTensors.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "train_RBM.py", line 82, in <module>
    trainRBM(args.data,args.learning_rate,args.k1,args.k2,args.epochs,args.batch_size,args.dimensions)
  File "train_RBM.py", line 49, in trainRBM
    rbm.persistive_contrastive_divergence_k(x_train)
  File "/home/cossio/deep-generative-models/src/obj/RBM.py", line 218, in persistive_contrastive_divergence_k
    g,m,r = self.adam(g,j,m,r)
  File "/home/cossio/deep-generative-models/src/obj/RBM.py", line 71, in adam
    if m == None and r == None:
  File "/home/cossio/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/math_ops.py", line 1336, in tensor_equals
    return gen_math_ops.equal(self, other)
  File "/home/cossio/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/gen_math_ops.py", line 3612, in equal
    name=name, ctx=_ctx)
  File "/home/cossio/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/gen_math_ops.py", line 3652, in equal_eager_fallback
    _attr_T, _inputs_T = _execute.args_to_matching_eager([x, y], _ctx)
  File "/home/cossio/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/eager/execute.py", line 261, in args_to_matching_eager
    ret = [internal_convert_to_tensor(t, dtype, ctx=ctx) for t in l]
  File "/home/cossio/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/eager/execute.py", line 261, in <listcomp>
    ret = [internal_convert_to_tensor(t, dtype, ctx=ctx) for t in l]
  File "/home/cossio/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py", line 1296, in internal_convert_to_tensor
    ret = conversion_func(value, dtype=dtype, name=name, as_ref=as_ref)
  File "/home/cossio/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/framework/constant_op.py", line 286, in _constant_tensor_conversion_function
    return constant(v, dtype=dtype, name=name)
  File "/home/cossio/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/framework/constant_op.py", line 227, in constant
    allow_broadcast=True)
  File "/home/cossio/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/framework/constant_op.py", line 235, in _constant_impl
    t = convert_to_eager_tensor(value, ctx, dtype)
  File "/home/cossio/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/framework/constant_op.py", line 96, in convert_to_eager_tensor
    return ops.EagerTensor(value, ctx.device_name, dtype)
ValueError: Attempt to convert a value (None) with an unsupported type (<class 'NoneType'>) to a Tensor.
```

**Comment by atreyasha** [comment]
Thank you for the feedback.

I had developed most of the code in with the beta version of TensorFlow 2.0, so some of these issues arise due to the changes since then.

This error arose because detections of `None` arguments was done as: `foo == None`. I changed this to `foo is None`.

This issue should be fixed with https://github.com/atreyasha/deep-generative-models/commit/e17b8ec1f7c6607b65fd3e5eee62051c1f1e9bbd

Let me know if it works fine.
